### DISCUSSION TITLE
Replace keras_tuner.engine.tuner to keras_tuner.Tuner

### DISCUSSION THREAD

**Discussion by AlekseyGur** [keras-team/autokeras#1895] [Q&A]
Hello!

It seems that method keras_tuner.engine.tuner was replaced/renamed by keras_tuner.Tuner.
Now it's necessary to make some changes  in this file: **/autokeras/engine/tuner.py**

Row № 30:
replace "class AutoTuner(keras_tuner.engine.tuner):" to "class AutoTuner(keras_tuner.Tuner):"

Am I right?

**Comment by AlekseyGur** [comment]
+ now during model.export_model() an error accrued in same file: **/autokeras/engine/tuner.py**:

Error: AttributeError: module 'keras_tuner.engine.tuner' has no attribute 'maybe_distribute'

Row № 62:
`with keras_tuner.engine.tuner.maybe_distribute(self.distribution_strategy):`
should be replaced with:
`with keras_tuner.src.engine.tuner.maybe_distribute(self.distribution_strategy):`

**Comment by haifeng-jin** [comment]
I have just updated KerasTuner to 1.4.3, which allows user to access from `keras_tuner.engine.tuner.Tuner`.
This is for backward compatibility. Just in case other packages also imported the tuner from this path.
So I just made it available.

**Reply by AlekseyGur** [reply]
Thank you!
One problem left with **maybe_distribute**

**Comment by AlekseyGur** [comment]
I don't understand why StructuredDataRegressor models have such a low prediction power.

My situation:
During training (using **x_train** and **y_train**) looses are quite low (MSE=50, val_loss=10). But if I try to use trained and exported model to predict **x_train** I get results that not even close to **y_train**. Can someone admit that situation? 

Seems like export_model return the worst underfitted model. Results differs very much. **y_train** contains values [-10, ... , 60] but predicted **y_train** contains values [-2, ... , +2]. 

In my dataset there are 22 features, 1700 examples, and simple regression performs good (good prediction power).

```Python
# Training
reg = ak.StructuredDataRegressor(
    overwrite=True,
    seed=42,
)
reg.fit(
    tf.data.Dataset.from_tensor_slices((x_train.astype(float), y_train.astype(float))),
    validation_split=0.25,
    verbose=1,
)
model = reg.export_model()

# Prediction
pred = model.predict(x_train.astype(float))
```

I tried to use x and y during fitting but results are the same: 

```Python
reg.fit(
    x=x_train.astype(float),
    y=y_train.astype(float),
    validation_split=0.25,
    verbose=1,
)
```

**Reply by AlekseyGur** [reply]
Oh, I,ve Just found a reason. I forgot to shuffle df before using it in a fit() method.
